backend/services/pos_service.py

```
@@ -0,0 +1,509 @@
+            parts = repo_url.rstrip('/').split('/')
+            owner = parts[-2] if len(parts) >= 2 else ""
+            repo = parts[-1] if parts else ""
+            
+            # GitHub API URL
+            api_url = f"https://api.github.com/repos/{owner}/{repo}/contents"
+            if commit_hash:
+                api_url += f"?ref={commit_hash}"
+            elif branch:
+                api_url += f"?ref={branch}"
+            
+            response = requests.get(api_url)
```

Ruff (0.16.2)

[error] Probable use of `requests` call without timeout

(S113)